nemo/collections/common/tokenizers/sentencepiece_tokenizer.py

```diff
@@ -242,7 +242,12 @@ def create_spt_model(
         for line in f:
             piece = line.split("\t")[0]
             token = piece[1:] if piece.startswith("▁") else f"##{piece}"
-            tokens.append(token)
+
+            if len(token) > 0:
```

Comment:
@yzhang123 Please review this patch to sentencepiece_tokenizer.py

Reply:
why are you doing this? cant we leave the empty token or just not add anything in case token is empty?

Reply:
`_` token is part of the tokenizer.vocab for the AN4 dataset. The above filtering `token = piece[1:] if piece.startswith("▁") else f"##{piece}"` causes a blank empty line to be added here if this check if not done. I have removed the addition of the token. Seems it doesnt affect AN4 dataset wer.

Reply:
Actually this doesnt work. I added back the check for asserting the vocabulary of the tokenizer and the models vocabulary, and the additional `_` token is necessary.